pyinstrument/renderers/profrenderer.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+import marshal
+from typing import Any, Callable
+
+from pyinstrument import processors
+from pyinstrument.frame import Frame
+from pyinstrument.renderers.base import FrameRenderer, ProcessorList
+from pyinstrument.session import Session
+
+# pyright: strict
+
+
+
+class ProfRenderer(FrameRenderer):
+    """
+    Outputs a marshaled dict, containing processed frames.
+    """
+
+    output_file_extension = "prof"
+
+    def __init__(self, **kwargs: Any):
+        super().__init__(**kwargs)
+
+    def render_frame(self, frame: Frame | None, stats: dict) -> dict:
+        if frame is None:
+            return {}
+
+        key = (frame.file_path or "", frame.line_no or 0, frame.function)
+        val = [-1, -1, frame.time, frame.total_self_time, {}]
+        if frame.parent:
+            p = frame.parent
+            val[-1][(p.file_path or "", p.line_no or 0, p.function)] = val[:-1]
+        if key in stats:
+            return
```

Comment:
The code in lines 36-38 below is not used. I thought there would be a way to collect better statistics, but none came to mind.

Reply:
would you mind removing them if so?